sink/README.md

```diff
@@ -40,17 +40,87 @@ a tarball to follow. This tarball will be extracted into the log directory.
+services like IRC or GitHub. The following fields might be present:
 
+ * `"onaborted"`: the status to process upon abortion
  * `"notify"`: message to send to IRC
  * `"github"`: object containing GitHub status info
  * `"badge"`: for updating badges
  * `"link"`: the link to the status message
  * `"extras"`: for putting extra files into the log directory
 
+### Aborted runs
+
+When the log ends without a final status line and the initial status
+contained a `"onaborted"` field, its value is used as the final
+status.
+
+Typically, the 'onaborted' value contains instructions to set a status
+on GitHub to "error", or to post a final message to IRC.
+
+If `"onaborted"` is not present, the sink will tweak to initial status
+and use that as the final status.  For example, it will modify the
+GitHub status to say "Aborted without status".
+
 ### Github Status format
 
-The `"github"` object above has the following fields:
+The `"github"` field of a status line can be used to make (almost)
+arbitrary REST requests against the GitHub API.  It can be used to
+update the status of commits and add comments to issues, for example.
+
+The following fields are defined:
+
+ * `"token"`: The OAuth token to use
+ * `"requests"`: A list of requests to perform
+
+Each request can have the following fields:
+
+ * `"method"`: The HTTP method to use, such as GET or POST
+ * `"resource"`: The resource to use, such as "/user"
+ * `"data"`: The data to send in the request
+ * `"result"`: A name for the result, to be used with string expansion
+
+The `"token"`, `"requests"`, and `"resource"` fields are mandatory.
+
+The `"resource"` and `"data"` values will be expanded.  Any occurance
+of "${path}" is replaced with a value from a previous request.  The
+'path' is a sequence of names, separated by "/" characters.  The first
+name is looked up among all the named results of previous requests,
+and the remaining names are then used to walk into that result.
+
+Here is an example:
+
+  { "github":
+    { "token": "......",
+      "requests": [
+        { "method": "GET",
+          "resource": "/user",
+          "result": "user"
+        },
+        { "method": "POST",
+          "resource": "repos/${user/login}/cockpit/issues",
+          "data": { "title": "New issue" }
+          "result": "issue"
+        },
+        { "method": "POST",
+          "resource": "${issue/comments_url}",
+          "data": { "body": "Very urgent" }
+        }
+      ]
+    }
+  }
+
+This will create a new issue in the "cockpit" repo of the
+authenticated user and add a comment to it.
+
+If a string is of the form "{path}" (i.e., it starts with "{" and ends
+with "}"), it is replaced completely by the value indicated by the
+path.  This might be string or some other JSON value such as a number.
+[ I don't know whether we need this, GitHub seems to be happy with
+strings instead of numbers, for example. ]
+
+For compatibility, the `"github"` object can also have the following
```

Comment:
I'd like to suggest we remove documentation for this, and remove support in the next month or so.